client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java

```diff
@@ -218,6 +222,24 @@ public void testGetJobStats() throws Exception {
         assertThat(exception.status().getStatus(), equalTo(404));
     }
 
+    public void testPostData() throws Exception {
+        String jobId = randomValidJobId();
+        Job job = buildJob(jobId);
+        MachineLearningClient machineLearningClient = highLevelClient().machineLearning();
+        machineLearningClient.putJob(new PutJobRequest(job), RequestOptions.DEFAULT);
+        machineLearningClient.openJob(new OpenJobRequest(jobId), RequestOptions.DEFAULT);
+
+        PostDataRequest postDataRequest = new PostDataRequest(jobId, XContentType.JSON);
+        for(int i = 0; i < 10; i++) {
+            Map<String, Object> hashMap = new HashMap<>();
+            hashMap.put("total", randomInt(1000));
+            postDataRequest.addDoc(hashMap);
```

Comment:
It would be better if these docs contained the job's time field, with the loop generating some ascending values for it.  Then the server could validate that they were received in ascending order.

To make this change an overload of `buildJob()` will have to be added that sets the time field and time format to specific values rather than random values, so that the loop knows what to add.